From 65212fdc6d4c1f798f12a69a0eb988167b5fa7ae Mon Sep 17 00:00:00 2001
From: MBL build <mbl@localhost>
Date: Wed, 30 Sep 2026 04:58:25 +0300
Subject: [PATCH 07/18] powerpc/44x: add AMCC apollo3g (WD My Book Live) board
 support

Original: ewaldc My-Book-Live kernel/patches/4.19/201-add-amcc-apollo3g-support.patch
---
 arch/powerpc/platforms/44x/Kconfig         | 11 +++++++++++
 arch/powerpc/platforms/44x/ppc44x_simple.c |  1 +
 2 files changed, 12 insertions(+)

diff --git a/arch/powerpc/platforms/44x/Kconfig b/arch/powerpc/platforms/44x/Kconfig
index 559577065af2..137008890b0a 100644
--- a/arch/powerpc/platforms/44x/Kconfig
+++ b/arch/powerpc/platforms/44x/Kconfig
@@ -120,6 +120,17 @@ config CANYONLANDS
 	help
 	  This option enables support for the AMCC PPC460EX evaluation board.
 
+config APOLLO3G
+	bool "Apollo3G"
+	depends on 44x
+	default n
+	select PPC44x_SIMPLE
+	select APM821xx
+	select IBM_EMAC_RGMII
+	select 460EX
+	help
+	  This option enables support for the AMCC Apollo 3G board.
+
 config GLACIER
 	bool "Glacier"
 	depends on 44x
diff --git a/arch/powerpc/platforms/44x/ppc44x_simple.c b/arch/powerpc/platforms/44x/ppc44x_simple.c
index 8d6e4da9dfbe..7a124f7e149d 100644
--- a/arch/powerpc/platforms/44x/ppc44x_simple.c
+++ b/arch/powerpc/platforms/44x/ppc44x_simple.c
@@ -50,6 +50,7 @@ machine_device_initcall(ppc44x_simple, ppc44x_device_probe);
  * board.c file for it rather than adding it to this list.
  */
 static char *board[] __initdata = {
+	"amcc,apollo3g",
 	"amcc,arches",
 	"amcc,bamboo",
 	"apm,bluestone",
